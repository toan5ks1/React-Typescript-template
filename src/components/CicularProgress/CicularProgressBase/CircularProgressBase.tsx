import { useMemo } from "react";

import { CircularProgress, CircularProgressProps } from "@mui/material";

export interface CircularProgressBaseProps extends CircularProgressProps {
    sizeVariant?: "button";
}

interface ICircularProgressVariantProps {
    size: CircularProgressProps["size"];
    color: CircularProgressProps["color"];
}

const CircularProgressBase = ({ sizeVariant, ...props }: CircularProgressBaseProps) => {
    const circularProgressVariantProps: ICircularProgressVariantProps = useMemo(() => {
        switch (sizeVariant) {
            case "button":
                return {
                    size: 18,
                    color: "primary",
                };
            default:
                return {
                    size: 40,
                    color: "primary",
                };
        }
    }, [sizeVariant]);

    return (
        <CircularProgress
            {...props}
            {...circularProgressVariantProps}
            data-testid="Loading__root"
        />
    );
};

export default CircularProgressBase;
