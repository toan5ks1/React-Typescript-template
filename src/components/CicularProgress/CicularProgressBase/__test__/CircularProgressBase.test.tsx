import CircularProgressBase, { CircularProgressBaseProps } from "../CircularProgressBase";

import { render, RenderResult } from "@testing-library/react";

describe("<CircularProgressBase />", () => {
    let wrapper: RenderResult;

    beforeEach(() => {
        wrapper = render(<CircularProgressBase />);
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});

describe("<CircularProgressBase />", () => {
    let wrapper: RenderResult;

    const props: CircularProgressBaseProps = {
        sizeVariant: "button",
    };

    beforeEach(() => {
        wrapper = render(<CircularProgressBase {...props} />);
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should have the height of 18px", () => {
        expect(wrapper.getByTestId("Loading__root")).toHaveStyle({
            height: "18px",
        });
    });
});
