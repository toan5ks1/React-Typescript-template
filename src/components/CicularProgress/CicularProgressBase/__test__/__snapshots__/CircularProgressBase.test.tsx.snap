// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularProgressBase /> should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-1 {
  display: block;
}

.emotion-2 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-0"
    data-testid="Loading__root"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg emotion-1"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-2"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;

exports[`<CircularProgressBase /> should match snapshot 2`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  display: inline-block;
  color: #1976d2;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-1 {
  display: block;
}

.emotion-2 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

<div>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-0"
    data-testid="Loading__root"
    role="progressbar"
    style="width: 18px; height: 18px;"
  >
    <svg
      class="MuiCircularProgress-svg emotion-1"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-2"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;
